modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/OpenIdConnectAuthenticationHandlerImpl.java

```diff
@@ -137,8 +138,36 @@ public void processAuthenticationResponse(
 			_getLoginRedirectURI(httpServletRequest),
 			oAuthClientEntry.getTokenRequestParametersJSON());
 
-		String userInfoJSON = _requestUserInfoJSON(
-			oidcTokens.getAccessToken(), oidcProviderMetadata);
+		URI userInfoURI = oidcProviderMetadata.getUserInfoEndpointURI();
+
+		String userInfoJSON = null;
+
+		if (userInfoURI == null) {
+				JWT idToken = oidcTokens.getIDToken();
+
+				JWTClaimsSet userInfoClaimSet = idToken.getJWTClaimsSet();
+
+				String email = userInfoClaimSet.getStringListClaim(
+					"emails").get(0);
```

Comment:
We need to avoid chaining method calls (coding convention).

So here you need to assign to a List<String> first.